packages/admin/admin-generator/src/commands/generate/generateForm/generateFormField.ts

```diff
@@ -478,50 +509,104 @@ export function generateFormField({
 
         formFragmentField = `${name} { id ${labelField} }`;
 
-        const filterConfig = config.filterField
+        const filterConfig = config.filter
             ? (() => {
-                  const filterField = findFieldByName(config.filterField.name, formConfig.fields);
-                  if (!filterField) {
+                  const filterField = config.filter.type === "field" ? findFieldByName(config.filter.name, formConfig.fields) : undefined;
+                  if (isComponentFormFieldConfig(filterField)) {
                       throw new Error(
-                          `Field ${String(config.name)}: No field with name "${
-                              config.filterField?.name
-                          }" referenced as filterField found in form-config.`,
+                          `Field ${String(config.name)}: referenced filter.type:"field" must not be type:"component". (This should never happen, because component type does not have name property.)`,
```

Comment:
If it should never happen, why add the check? To make TS happy?

Reply:
Yes, exactly.
And I thought it would at least be a good idea to describe why this shouldn't happen, in case name is later added to component-type. But on second thought, I’m not sure referencing component-type makes sense in this context at all.

Reply:
I'd add the "This should never happen..." part as comment above the throw.